core/base/src/TROOT.cxx

```diff
@@ -2574,14 +2544,258 @@ const char**& TROOT::GetExtraInterpreterArgs() {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+
+#ifdef ROOTPREFIX
+static Bool_t IgnorePrefix() {
+   static Bool_t ignorePrefix = kFALSE;
+   static Bool_t init = kFALSE;
+   if (!init) {
+      if (gSystem->Getenv("ROOTIGNOREPREFIX"))
+         ignorePrefix = kTRUE;
+      init = kTRUE;
+   }
+   return ignorePrefix;
+}
+#endif
+
+////////////////////////////////////////////////////////////////////////////////
+/// Get the rootsys directory in the installation. Static utility function.
+
+const TString& TROOT::GetRootSys() {
+#ifdef ROOTPREFIX
+   if (IgnorePrefix()) {
```

Comment:
Could `IgnorePrefix()` return `false` if `ROOTPREFIX` is not defined? Then the logic here (and below) should become much simpler. Then maybe rename `IgnorePrefix()` into `UsePrefix()` and invert the logic.